---
[v1] net: ethernet: mtk_eth_soc: fix mtk_eth_offload_init memory leak

Fix memory leak in mtk_eth_offload_init by correcting initialization sequence.
This moves offload initialization outside the PPE loop to prevent multiple
allocations, ensuring proper resource management.
---
v1: Fix mtk_eth_offload_init memory leak.
---

From 970222d6ae0073e38289a691245a0f13392b8641 Mon Sep 17 00:00:00 2001
From: "chak-kei.lam" <chak-kei.lam@mediatek.com>
Date: Tue, 3 Feb 2026 10:59:49 +0800
Subject: [PATCH] net: ethernet: mtk_eth_soc: fix mtk_eth_offload_init memory

---
 drivers/net/ethernet/mediatek/mtk_eth_soc.c | 8 ++++----
 drivers/net/ethernet/mediatek/mtk_eth_soc.h | 2 +-
 2 files changed, 5 insertions(+), 5 deletions(-)

--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.c
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.c
@@ -6164,7 +6164,7 @@ static int mtk_probe(struct platform_dev
 				err = -ENOMEM;
 				goto err_deinit_ppe;
 			}
-			err = mtk_eth_offload_init(eth, i);
+			err = mtk_eth_offload_init(eth);
 
 			if (err)
 				goto err_deinit_ppe;
--- a/drivers/net/ethernet/mediatek/mtk_eth_soc.h
+++ b/drivers/net/ethernet/mediatek/mtk_eth_soc.h
@@ -1691,7 +1691,7 @@ int mtk_gmac_gephy_path_setup(struct mtk
 int mtk_gmac_rgmii_path_setup(struct mtk_eth *eth, int mac_id);
 int mtk_gmac_usxgmii_path_setup(struct mtk_eth *eth, int mac_id);
 
-int mtk_eth_offload_init(struct mtk_eth *eth, u8 id);
+int mtk_eth_offload_init(struct mtk_eth *eth);
 int mtk_eth_setup_tc(struct net_device *dev, enum tc_setup_type type,
 		     void *type_data);
 int mtk_flow_offload_cmd(struct mtk_eth *eth, struct flow_cls_offload *cls,
